core/crypto/pedersen_hash.go

```diff
@@ -77,15 +72,14 @@ func Pedersen(a *felt.Felt, b *felt.Felt) *felt.Felt {
 
 func processElement(a *fp.Element, p1 *starkcurve.G1Jac, p2 *starkcurve.G1Jac) *starkcurve.G1Jac {
 	var bigInt big.Int
-	aRegular := a.ToRegular()
-
-	lowPart := aRegular
-	lowPart[3] = lowPart[3] & lowPartMask // Zero-out the top nibble (bits 249-252)
-	m := new(starkcurve.G1Jac).ScalarMultiplication(p1, lowPart.ToBigInt(&bigInt))
+	var aBytes [32]byte
+	a.BigInt(&bigInt).FillBytes(aBytes[:])
 
-	highPart := bigInt.SetUint64(aRegular[3] >> lowPartBits) // The top nibble (bits 249-252)
+	highPart := bigInt.SetUint64(uint64(aBytes[0])) // The top nibble (bits 249-252)
+	lowPart := aBytes[1:]                           // Zero-out the top nibble (bits 249-252)
 
+	m := new(starkcurve.G1Jac).ScalarMultiplication(p2, highPart)
 	var n starkcurve.G1Jac
-	n.ScalarMultiplication(p2, highPart)
+	n.ScalarMultiplication(p1, bigInt.SetBytes(lowPart))
```

Comment:
I don't know too much about elliptical curves but is the order of multiplication not important?

Reply:
Nope, `a + b = b + a`